File: 7KYU/Maximum Length Difference/solution.py
def mxdiflg(a1, a2):
    if len(a1) == 0 or len(a2) == 0:
        return -1
    a1_counts = [len(i) for i in a1]
    a2_counts = [len(i) for i in a2]
    
    a1_max = max(a1_counts)
    a2_max = max(a2_counts)
    a1_min = min(a1_counts)
    a2_min = min(a2_counts)
    
    max_diff_a1_a2 = a1_max - a2_min
    max_diff_a2_a1 = a2_max - a1_min
    
    if max_diff_a1_a2 > max_diff_a2_a1:
        return max_diff_a1_a2
    return max_diff_a2_a1

File: 7KYU/Maximum Length Difference/solution.cs
using System;
using System.Linq;

public class MaxDiffLength 
{
    
    public static int Mxdiflg(string[] a1, string[] a2) 
    {
        if(a1.Length == 0 || a2.Length == 0)
            return -1;
        var a1Max = a1.Select(s => s.Length).Max();
        var a1Min = a1.Select(s => s.Length).Min();
        var a2Min = a2.Select(s => s.Length).Min();
        var a2Max = a2.Select(s => s.Length).Max();

        var a1MaxMinusA2Min = Math.Abs(a1Max - a2Min);
        var a2MaxMinusA1Min = Math.Abs(a1Min - a2Max);
        return Math.Max(a1MaxMinusA2Min, a2MaxMinusA1Min);
    }
}